### PR TITLE
#83 added updates to the checker

### PR DESCRIPTION
**Pull request by calchoo**
Add changes to checker to handle non mavis output directories. Also has returns non zero exit status if failure detected.

### CODE REVIEW

**Comment by creisle** [comment]
Can you add some test for this new functionality

File: tests/integration/test_checker.py [hunk 1]
```diff
@@ -1,9 +1,15 @@
...
 import unittest
-from unittest.mock import patch
-from unittest.mock import mock_open
-
+from unittest.mock import mock_open, patch
 
 from mavis import checker
+from mavis.constants import COMPLETE_STAMP, SUBCOMMAND
+
+DATA_PREFIX = os.path.join(os.path.dirname(__file__), 'data')
```
Comment (creisle) [inline tests/integration/test_checker.py]: you can just import this from the package i think?
```
from . import DATA_PREFIX
```

File: tests/integration/test_checker.py [hunk 2]
```diff
@@ -86,3 +92,50 @@ def test_parse_valid_run_time(self):
             isfile.return_value = True
             result = checker.parse_run_time('log')
         self.assertEqual(1, result)
+
+
+class TestCompletion(unittest.TestCase):
```
Comment (creisle) [inline tests/integration/test_checker.py]: you are missing tests for when individual steps fail

Comment (calchoo) [reply tests/integration/test_checker.py]: I've added some more tests

**Pull request merged**